OptimizelySDKCore/OptimizelySDKCore/OPTLYBaseCondition.m

```diff
@@ -181,7 +186,7 @@ - (nullable NSNumber *)evaluateMatchTypeLessThan:(NSDictionary<NSString *, NSObj
         // Log Invalid Attribute Value Type
```

Comment:
Actually, I do have one more request, which is unfortunately somewhat complicated.

Can we split up `isValidNumericAttributeValue` into two checks?
1. `isNumericAttributeValue`, which only checks the type of the passed value, with the existing log message `OPTLYLoggerMessagesAudienceEvaluatorConditionEvaluatedAsUnknownForUnexpectedTypeNull`.
2. `isFiniteNumber`, which checks makes sure the number isn't NaN/inf/large, and whose error message is `@"[AUDIENCE EVALUATOR] Audience condition %@ evaluated to UNKNOWN because the number value for user attribute %@ is not in the range [-2^53, +2^53]."`

The reasoning is that if a customer passes in a value like NaN, infinity, or 9007199254740992, they probably wouldn't understand if we complain about the data type. We need to inform them that the number is out of bounds.

(This applies to the `exact`, `gt`, and `lt` matchers.)